File: GameServer/Models/Skills/Skill.cs
using GameServer.Models.Attributes;
namespace GameServer.Models.Skills
{
    public class Skill : IModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal BaseDamage { get; set; }
        public List<AttributeMultiplier> DamageMultipliers { get; set; }
        public string Description { get; set; }
        public int CooldownMS { get; set; }
        public string IconPath { get; set; }

        public Skill(GameCore.Entities.Skill skill)
        {
            Id = skill.Id;
            Name = skill.Name;
            BaseDamage = skill.BaseDamage;
            DamageMultipliers = skill.SkillDamageMultipliers.Select(x => new AttributeMultiplier(x)).ToList();
            Description = skill.Description;
            CooldownMS = skill.CooldownMS;
            IconPath = skill.IconPath;
        }
    }
}

File: GameServer/TypeScript/Game/new-svelte/src/lib/battle/skill.ts
import { Tooltippable } from "$lib/tooltips";
import { ISkill, IAttributeMultiplier } from "$lib/api";
import { Battler } from "./battler";

export class Skill implements ISkill {
    id: number
    name: string;
    chargeTime: number;
    baseDamage: number;
    damageMultipliers: IAttributeMultiplier[];
    description: string;
    cooldownMS: number;
    iconPath: string;
    owner: Battler;
    ownerStatsVersion = -1;
    target?: Battler;

    constructor(data: ISkill, owner: Battler) {
        this.id = data.id
        this.name = data.name;
        this.baseDamage = data.baseDamage;
        this.damageMultipliers = data.damageMultipliers;
        this.description = data.description;
        this.cooldownMS = data.cooldownMS;
        this.chargeTime = 0;
        this.iconPath = data.iconPath;
        this.owner = owner;
    }

    calculateDamage() {
        let dmg = this.baseDamage;
        this.damageMultipliers.forEach((dmgType) => {
            dmg += this.owner.attributes.getValue(dmgType.attributeId) * dmgType.multiplier;
        })
        return dmg
    }
}
